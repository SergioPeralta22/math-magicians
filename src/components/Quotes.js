import { useEffect, useState } from 'react';

function Quotes() {
  const [quotes, setQuotes] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  // usually I would put this in a .env file but I'm not sure
  // if the reviewer will be able to see it.
  const API_ID = 'dByEAwV9LJFlCKUgjAX60A==fvZH8LP91QrzRc6I';
  const URL = 'https://api.api-ninjas.com/v1/quotes';

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const response = await fetch(URL, {
          headers: {
            'X-Api-Key': API_ID,
          },
        });
        const data = await response.json();
        setQuotes(data[0]);
        setLoading(false);
      } catch (error) {
        setError('Error fetching quote');
        setLoading(false);
      }
    };
    fetchQuote();
  }, []);

  return (
    <div className="quotes-container">
      {loading && <div className="loading">Loading...</div>}
      {error && (
      <div className="error">
        {error}
        - Please try again later
      </div>
      )}
      {quotes && (
      <p className="quote">
        &quot;
        {quotes.quote}
        {' '}
        &quot; -
        <span>
          {' '}
          {quotes.author}
        </span>
      </p>
      )}
    </div>
  );
}

export default Quotes;
