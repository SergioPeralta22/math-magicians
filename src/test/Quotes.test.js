import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../components/Quotes';

describe('Quotes component', () => {
  test('Renders correctly', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });

  test('Renders loading text', () => {
    const { getByText } = render(<Quotes />);
    const loading = getByText(/loading/i);
    expect(loading).toBeInTheDocument();
  });
});
