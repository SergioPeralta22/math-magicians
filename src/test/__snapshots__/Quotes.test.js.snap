// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component Renders correctly 1`] = `
<div>
  <div
    class="quotes-container"
  >
    <div
      class="loading"
    >
      Loading...
    </div>
  </div>
</div>
`;
